src/cgnslib.c

```diff
@@ -13528,226 +13644,1865 @@ int cg_1to1_average_write(int fn, int B, int Z, int J,
- *                 you may also use the \e "Dirichlet", \e 0 or \e "Neuman", \e 0 to access the node. See
- *                 \ref CGNS_Navigation_Ill "CGNS Navigation Illustration" for example usage.
+ * \param[in] fn \FILE_fn
+ * \param[in] B   \B_Base
+ * \param[in] particlename   Name of the particle zone.
+ * \param[in] size  Number of particles in this particle zone
+ * \param[out] P  \P_ParticleZone
  * \return \ier
- *
- * \details The character string "end" (or 'end' for the Fortran function) must be the last argument.
- *          It is used to indicate the end of the argument list. You may also use the empty string,
- *          "" ('' for Fortran), or the NULL string in C, to terminate the list.
  */
-int cg_gorel(int fn, ...)
+int cg_particle_write(int fn, int B, const char *particlename, const cgsize_t size, int *P)
 {
-    va_list ap;
-    int status;
-    va_start (ap, fn);
-    status = vcg_gorel(fn, ap);
-    va_end(ap);
-    return status;
-}
+   cgns_base *base;
+   cgns_pzone *pzone = NULL;
+   int index;
+   cgsize_t dim_vals[1];
 
-/*-----------------------------------------------------------------------
- *              F2008 C-FORTRAN INTERFACE ROUTINE
- *
- *      cg_gorel function which is compatible with F2008 and TS 29113
- *      "Further Interoperability of Fortran with C WG5/N1942"  and
- *      allows optional function parameters to be passed to a C function
- *      which has variable number of arguments. This function is
- *      directly callable from FORTRAN.
- *
- */
-int cg_gorel_f08(int fn, ...)
-{
-    int n = 0;
-    int index[CG_MAX_GOTO_DEPTH];
-    char *label[CG_MAX_GOTO_DEPTH];
-    va_list ap;
+   /* verify input */
+   if (cgi_check_strlen(particlename)) return CG_ERROR;
 
-    if (posit == 0) {
-        cgi_error ("position not set with cg_goto");
-        return CG_ERROR;
-    }
-    if (fn != posit_file) {
-        cgi_error("current position is in the wrong file");
-        return CG_ERROR;
-    }
+   /* get memory address file */
+   cg = cgi_get_file(fn);
+   if (cg == 0) return CG_ERROR;
 
-    va_start (ap, fn);
-    for (n = 0; n < CG_MAX_GOTO_DEPTH; n++) {
-        label[n] = va_arg(ap, char *);
-        if (label[n] == NULL || label[n][0] == 0) break;
-        if (strcmp("end",label[n])==0 || strcmp("END",label[n])==0) break;
-        index[n] = (int)*va_arg(ap, int *);
-    }
-    va_end(ap);
+   /* verify input */
+   if (cgi_check_mode(cg->filename, cg->mode, CG_MODE_WRITE)) return CG_ERROR;
 
-    return cgi_update_posit(n, index, label);
-}
+   /* get memory address for base */
+   base = cgi_get_base(cg, B);
+   if (base==0) return CG_ERROR;
 
-/*-----------------------------------------------------------------------*/
-/**
- * \ingroup AccessingANode
- *
- * \brief Access a node via pathname
- *
+   /* Verify the particle count - We allow 0 sized ParticleZone_t nodes but they cannot have and solution/coordinate arrays */
```

Comment:
Verify the particle count - we allow 0-sized ParticleZone_t nodes, but they cannot have solution/coordinate arrays

Reply:
Good catch, fixed